main.js

```diff
@@ -0,0 +1,285 @@
+  }
+};
+
+// Fetch current weatehr data for the location queried
+const fetchNow = async () => {
+  const query = document.querySelector("#query").value.replace(/\s+/g, "%20");
+
+  const url = `https://api.openweathermap.org/data/2.5/weather?q=${query}$&limit=1&appid=${apiKey}&units=${units}`;
+
+  const fetchedData = await fetch(url, {
+    method: "Get",
+    dataType: "json",
+  });
+
+  const nowData = await fetchedData.json();
+
+  return nowData;
+};
+
+// Use template to update HTML for the current weather forecast based on data from the queried location
+const addNow = async (data) => {
+  const nowSection = document.querySelector("#now");
+  nowSection.replaceChildren();
+
+  const template = `      
+    <div class="row justify-content-center forecast-today my-3" id="now">
+      <div class="col-md-6">
+        <h3 class="text-center">${data.main.temp.toFixed(1)}°</h3>
+        <h4 class="text-center">${data.name}</h4>
+        <h5 class="text-center">${data.weather[0].main}</h5>
+      </div>
+      <div class="col-md-6">
+        <img
+          class="img-fluid mx-auto d-block"
+          src="https://openweathermap.org/img/wn/${data.weather[0].icon}@2x.png" )
+          alt="unable to load image"
+        />
+      </div>
+    </div>
+  `;
+
+  nowSection.insertAdjacentHTML("beforeend", template);
+};
+
+// Use a series of functions to populate the 5-day forecast section of the HTML
+const getFiveDay = async () => {
+  try {
+    const coordinates = await fetchCoordinates();
+    const fiveDayData = await fetchFiveDayData(coordinates);
+    const processedData = await processFiveDayData(fiveDayData);
+    addFiveDay(processedData);
+  } catch (error) {
+    console.error("Error processing data:", error);
+  }
+};
+
+// Use Open Weather Geolocation API to return data, {lat:____, lon:___}, for the queried location
+const fetchCoordinates = async () => {
+  const query = document.querySelector("#query").value.replace(/\s+/g, "%20");
+
+  const coordinatesCall = await fetch(
+    `https://api.openweathermap.org/geo/1.0/direct?q=${query}&limit=1&appid=${apiKey}
+    `,
+    {
+      method: "GET",
+      dataType: "json",
+    }
+  );
+
+  const coordinatesArray = await coordinatesCall.json();
+  const coordinates = coordinatesArray[0];
+
+  return coordinates;
+};
+
+// Given an object with {lat:___, lon:___} syntax, fetch five-day weather information. This returns 40 instances of data which must be divided into 5 sets of 8.
+const fetchFiveDayData = async (coordinates) => {
+  const fiveDayURL = `https://api.openweathermap.org/data/2.5/forecast?lat=${coordinates.lat}&lon=${coordinates.lon}&appid=${apiKey}&units=${units}`;
+
+  const fetchFiveDay = await fetch(fiveDayURL, {
+    method: "GET",
+    dataType: "json",
+  });
+
+  const fiveDayData = await fetchFiveDay.json();
+
+  return fiveDayData;
+};
+
+// Colate data from each day in the five-day forecast.
+const processFiveDayData = async (data) => {
```

Comment:
this function is still complex, nested and logic that helpers can help with readability and debugging